### PR TITLE
feat: add batchnorm kernel

### PR DESCRIPTION
**Pull request by 0xrushi**
```
❯ python main.py batchnorm_benchmark --benchmark=True
Running BatchNorm benchmark: 512x2048, use_triton=True
Output shape: torch.Size([512, 2048]), dtype: torch.float32
Benchmark results
                             kernel_path  non_triton    triton  non_triton-triton
kernel                                                                           
batchnorm_benchmark  batchnorm_benchmark     0.00917  0.000153           0.009017

==================================
```

```
❯ python main.py batchnorm_benchmark --benchmark=True --batch_size=1024 --channels=4096
Running BatchNorm benchmark: 1024x4096, use_triton=True
Output shape: torch.Size([1024, 4096]), dtype: torch.float32
Benchmark results
                             kernel_path  non_triton    triton  non_triton-triton
kernel                                                                           
batchnorm_benchmark  batchnorm_benchmark    0.008819  0.000216           0.008603

==================================
```

### CODE REVIEW

**Review by adamomainz — CHANGES_REQUESTED** [review]
This really seems more suited for tritonbench
https://github.com/meta-pytorch/tritonbench

If you move it there I would be happy to review
